### PR TITLE
Fix hosts links rendering

### LINKED ISSUE #3848: Hosts view renders empty page: 'TypeError: Cannot read property 'length' of undefined'
build **Nightly**

build time **201807100044**

```
$ chronograf --version
2018/07/10 09:03:02 Chronograf 1.4.3.0~n201807100044 (git: 4f2378a8a0147a1187d86fd78747ade306663f8d)
```
**Use Case**

1. Open the Hosts tab
2. From the Hosts list click on a Host

The page does not render and there is a message "A Chronograf error has occurred. Please report the issue here."

The console log contains an error:

```
...
app.3c6a9d84521678057a87.js:210 TypeError: Cannot read property 'length' of undefined
...
```

**Expected Behavior**

The page should show graphs from telegraf system time series. 

**Screenshots**

hostviewmissing01
hostviewmissing02

**Console log**
```
app.3c6a9d84521678057a87.js:210 TypeError: Cannot read property 'length' of undefined
    at t.get (app.3c6a9d84521678057a87.js:218)
    at t.get (app.3c6a9d84521678057a87.js:218)
    at t.value (app.3c6a9d84521678057a87.js:218)
    at h (app.3c6a9d84521678057a87.js:210)
    at beginWork (app.3c6a9d84521678057a87.js:210)
    at a (app.3c6a9d84521678057a87.js:210)
    at o (app.3c6a9d84521678057a87.js:210)
    at E (app.3c6a9d84521678057a87.js:210)
    at b (app.3c6a9d84521678057a87.js:210)
    at _ (app.3c6a9d84521678057a87.js:210)
br @ app.3c6a9d84521678057a87.js:210
commitErrorLogging @ app.3c6a9d84521678057a87.js:210
T @ app.3c6a9d84521678057a87.js:210
E @ app.3c6a9d84521678057a87.js:210
b @ app.3c6a9d84521678057a87.js:210
_ @ app.3c6a9d84521678057a87.js:210
interactiveUpdates @ app.3c6a9d84521678057a87.js:210
En @ app.3c6a9d84521678057a87.js:210
app.3c6a9d84521678057a87.js:1 TypeError: Cannot read property 'length' of undefined
    at t.get (app.3c6a9d84521678057a87.js:218)
    at t.get (app.3c6a9d84521678057a87.js:218)
    at t.value (app.3c6a9d84521678057a87.js:218)
    at h (app.3c6a9d84521678057a87.js:210)
    at beginWork (app.3c6a9d84521678057a87.js:210)
    at a (app.3c6a9d84521678057a87.js:210)
    at o (app.3c6a9d84521678057a87.js:210)
    at E (app.3c6a9d84521678057a87.js:210)
    at b (app.3c6a9d84521678057a87.js:210)
    at _ (app.3c6a9d84521678057a87.js:210)
value @ app.3c6a9d84521678057a87.js:1
commitErrorLogging @ app.3c6a9d84521678057a87.js:210
T @ app.3c6a9d84521678057a87.js:210
E @ app.3c6a9d84521678057a87.js:210
b @ app.3c6a9d84521678057a87.js:210
_ @ app.3c6a9d84521678057a87.js:210
interactiveUpdates @ app.3c6a9d84521678057a87.js:210
En @ app.3c6a9d84521678057a87.js:210
app.3c6a9d84521678057a87.js:1 {componentStack: "↵    in t↵    in div↵    in t↵    in n↵    in Conn… in RouterContext↵    in Router↵    in t↵    in t"}
value @ app.3c6a9d84521678057a87.js:1
commitErrorLogging @ app.3c6a9d84521678057a87.js:210
T @ app.3c6a9d84521678057a87.js:210
E @ app.3c6a9d84521678057a87.js:210
b @ app.3c6a9d84521678057a87.js:210
_ @ app.3c6a9d84521678057a87.js:210
interactiveUpdates @ app.3c6a9d84521678057a87.js:210
En @ app.3c6a9d84521678057a87.js:210
```

### PR DESCRIPTION
**Pull request by OfTheDelmer**
Closes #3848

_What was the problem?_
The `HostPage` was rendering a `DashboardHeader` with incorrect props after #3828.   
_What was the solution?_
Update the `HostPage` props passed to `DashboardHeader` and add a test utility to fetch and format host links. 

Additionally, dead code was deleted for `filteredHosts`. This was something that was happening before `getAppsForHosts` was replaced by `getAppsForHost` where `getAppsForHosts` uses mutation on a `hosts` to [add `apps` and `tags`](https://github.com/influxdata/chronograf/blob/ac6de1c671fd7858d29276b065a0c96a48f35efd/ui/src/hosts/apis/index.js#L186-L192)  to each host fetched from `getAllHosts`.  

  - [x] Rebased/mergeable
  - [x] Tests pass

### CODE REVIEW

File: ui/src/hosts/utils/hostsSwitcherLinks.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      text: h.name,
+      to: `/sources/${source.id}/hosts/${h.name}`,
+    }
+  })
+
+  return {links, active: null}
+}
+
+export const updateActiveHostLink = (
+  dashboardLinks: DashboardSwitcherLinks,
```
Comment (OfTheDelmer) [inline ui/src/hosts/utils/hostsSwitcherLinks.ts]: I can name this `hostLinks`

File: ui/src/hosts/utils/hostsSwitcherLinks.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import {DashboardSwitcherLinks} from 'src/types/dashboards'
+
+export const EMPTY_LINKS = {
+  links: [],
+  active: null,
+}
+
+const hostNamesAJAX = getAllHostsAJAX as GetAllHosts
+
+export const loadHostsLinks = async (
```
Comment (121watts) [inline ui/src/hosts/utils/hostsSwitcherLinks.ts]: This should go in an `api` dir

File: ui/src/hosts/utils/hostsSwitcherLinks.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import {Source} from 'src/types/sources'
+import {HostNames, HostName} from 'src/types/hosts'
+import {DashboardSwitcherLinks} from 'src/types/dashboards'
+
+export const EMPTY_LINKS = {
+  links: [],
+  active: null,
+}
+
+const hostNamesAJAX = getAllHostsAJAX as GetAllHosts
```
Comment (121watts) [inline ui/src/hosts/utils/hostsSwitcherLinks.ts]: Try not to force typing here.  `const hostNamesAJAX: GetAllHosts = getAllHostsAJAX`

File: ui/test/hosts/utils/hostsSwitcherLinks.test.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import {
```
Comment (121watts) [inline ui/test/hosts/utils/hostsSwitcherLinks.test.ts]: Nice!

**Pull request merged**